src/lib/app_config.js

```diff
@@ -142,14 +144,19 @@ class AppConfig {
     }
 
     /**
-     * @param {string} string
+     * @param {string} key
+     * @param {string} defaultString
      * @param {{ [x: string]: any; }} [params]
      */
-    nls(string, params) {
-        if (messages && !messages[string])  {
-            warn("No message found for '" + string + "' in the provided messages, falling back to default English message.");
-        }
-        var translated = messages && messages[string] || string;
+    nls(key, defaultString, params) {
+        if (!messages[key])  {
+            warn("No message found for the key '" + key + "' in the provided messages, trying to find a translation for the default string '" + defaultString + "'.");
+            if (!messages[defaultString]) {
```

Comment:
are we keeping this default string logic for backwards compatibility reasons?

Reply:
yeah, I kept this so that Ace users who currently have translations set-up don't get those broken by this change (well that's the goal at least 😄)